.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.10
+  VCPKG_COMMIT: 32eccc18191fbb57b159784a1724d2d00613ae82
+  VCPKG_DEST_MACOS: /Users/runner/qbt_tools/vcpkg
+  VCPKG_DEST_WIN: C:\qbt_tools\vcpkg
+  UBUNTU_LIBTORRENT_VERSION: libtorrent-1.2.10
+
+# NOTE: for client code built with >= C++14, libtorrent 1.2.x must be built with >= C++14 as well.
+# Compilers that don't default to >= C++14 must be overriden.
+# See https://github.com/arvidn/libtorrent/issues/4965.
+
+jobs:
+
+  ci_ubuntu:
+    name: Ubuntu
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-18.04]
```

Comment:
Since the binary produced aren't portable, there is no sense to build in `ubuntu-18.04`.

Reply:
@Chocobo1 

We need to build in `ubuntu-18.04`, to make sure no changes accidentally break support for it. When we decide that we're going to drop support for it, we'll do so as a conscious decision, and remove it from CI accordingly.

As for the binaries not being portable, that is not an issue, see the earlier comment about that: https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491119532.

Reply:
>We need to build in ubuntu-18.04, to make sure no changes accidentally break support for it.

This sounds like we are supporting/targeting a specific distro version, but I can tell you, we do not.
We only list out the required lib versions for users/distros to follow. We also take into account the library version availability of the distros when bumping our lib requirements.

>When we decide that we're going to drop support for it, we'll do so as a conscious decision, and remove it from CI accordingly.

Wrong way of reasoning, if that were the case, we should be building & testing for every distro out there which is wrong.

Reply:
@Chocobo1 

>This sounds like we are supporting/targeting a specific distro version, but I can tell you, we do not.
We only list out the required lib versions for users/distros to follow. We also take into account the library version availability of the distros when bumping our lib requirements.

The most important distro to consider when evaluating library availablity is Ubuntu LTS (and also Debian stable, to an extent).
I was under the impression that we support the current LTS Ubuntu releases, if not the full 5 years, at least an honest attempt at that (3 years, which is what the flavors support, is good enough).
How can we at lest not have that commitment?

We still support Windows 7, and it has been hell to argue for dropping support for it.
How does that make any sense?? The bug handling team is already strained with having to deal with weird heisenbugs and issues with 10+ year old EOL proprietary OSes, that don't even have support for basic stuff like paths over 256 characters, and yet we can throw users of the most popular Linux LTS distro under the bus at any moment?
Seems like the priorities for OS support are all wrong.
Same for the WebUI still supporting IE 11 (in a practical sense, and also in the sense that there is an official commitment to that: https://github.com/qbittorrent/qBittorrent/blob/master/src/webui/www/README.md).

> Wrong way of reasoning, if that were the case, we should be building & testing for every distro out there which is wrong.

That's an implication that is not present in what I said. At the very least we should have CI for the most important configurations, which is what this PR achieves.
And of course, if you want to add support CI for more distros (the more the better), that can be done later.

Reply:
>We still support Windows 7, and it has been hell to argue for dropping support for it.

It only shows your opinion is still (currently) the less popular one, on user side and among the developers.
You're getting derailed into OS support discussion... that is not the point here.

So you're installing some official packages from ubuntu, does the versions that offered in ubuntu-20.04 and ubuntu-18.04 differ that much?

Reply:
@Chocobo1 

> So you're installing some official packages from ubuntu, does the versions that offered in ubuntu-20.04 and ubuntu-18.04 differ that much?

The main problem is boost and Qt:

- boost:
    - 18.04: [1.65.1.0ubuntu1](https://packages.ubuntu.com/bionic/libboost-all-dev)
    - 20.04: [1.71.0.0ubuntu2](https://packages.ubuntu.com/focal/libboost-all-dev)

- Qt:
    - 18.04: [5.9.5+dfsg-0ubuntu2.5](https://packages.ubuntu.com/bionic-updates/qt5-default)
    - 20.04: [5.12.8+dfsg-0ubuntu1](https://packages.ubuntu.com/focal/qt5-default)

> It only shows your opinion is still (currently) the less popular one, on user side and among the developers.
You're getting derailed into OS support discussion... that is not the point here.

Sad that apparently "popularity" trumps good maintenance practices and is more important than the bug reporting team. It's not even as popular as you might think. There's always a very vocal minority of semi tech illiterate lunatics holding on to old Windows versions unreasonably long because of "nostalgia" or "it worked better"/"I don't like the one" (without providing valid objective reasons) or "the new version spies more" (lol). Just look at how many people held on to Windows XP well past what was reasonable. Note that this is of course very different from keeping a secondary installation around (sometimes in a VM) for preservation purposes of some software/hardware that doesn't run reasonably anywhere else. My problem is when people make such arguments for their daily driver machines. If they want to run museum piece OSes, they can use older qBittorrent versions. They just won't be able to install new ones.

Let me be clear: I don't want to deal with issues relating to Windows < 10 any longer (and I'm sure other contributors/active users who help solve Windows issues don't either), and I don't think the lunatic vocal minority is more important than me or my contributions to the project. The team is already small enough. We have to make compromises. This is yet another thing that unnecessarily clutters the issue tracker and increases maintenance burden.

Reply:
>18.04: 5.9.5+dfsg-0ubuntu2.5
20.04: 5.12.8+dfsg-0ubuntu1

Now I think it is worth to test in both OS versions, 18.04 for our minimum Qt requirement and 20.04 for Qt LTS (obsoleted though). And I also suppose when we raise Qt requirement to >= 5.12 we can drop building in 18.04.

>Sad that apparently "popularity" trumps good maintenance practices and is more important than the bug reporting team.... 

You presented your thoughts, OK.

Reply:
> And I also suppose when we raise Qt requirement to >= 5.12 we can drop building in 18.04.

Hopefully, not in the near future.